agent/lib/kontena/workers/weave_worker.rb

```diff
@@ -48,19 +48,21 @@ def on_dns_add(topic, event)
     # @param [String] topic
     # @param [Docker::Event] event
     def on_container_event(topic, event)
-      if event.status == 'start'
+      if network_adapter.router_image?(event.from)
+        if event.status == 'start' || event.status == 'restart'
```

Comment:
Seems like both restart and re-create cases will include a `start` event, so let's just stick to that.